.github/workflows/main.yml

```diff
@@ -0,0 +1,128 @@
+name: CI
+
+on:
+  push:
```

Comment:
I couldn't quite work out if a merge would cause a `push` trigger to run on `master`. It seems like the cache is only accessible from `master` -> `featureBranch`, so we'd need a build to occur on `master` to get the benefit in an initial `pull_request` build.

Reply:
I think a merge should trigger a `push` event, but I'm still not clear as to whether it's sufficient for `main.yml` to be in `master` before merge, or be part of the merge. For testing, `push` triggers made sure every push to my branch triggered the workflow. I was hoping this PR would trigger the workflow for the PR itself, but now I'm assuming it will need to be merged first.

Reply:
Yeah, that makes sense. Either way, I don't think its an issue for this PR, as we can always add nightlies or some such to keep a cache available from `master` for PRs in the future.